pointcloud/basic_octree_test.go

```diff
@@ -598,3 +598,78 @@ func testPCDToBasicOctree(t *testing.T, artifactPath string) {
 
 	validateBasicOctree(t, basicOct, basicOct.center, basicOct.sideLength)
 }
+
+func TestCachedMaxChildProbability(t *testing.T) {
```

Comment:
[major] I would add this checking process to the `validateBasicOctree` function as it goes through and checks the metadata associated with each node. By adding it to that function, this test will be performed everywhere instead of just this one test case.